### PR TITLE
refactor(expressions): move generic methods to ValueExpr class

### PR DESCRIPTION
**Pull request by cpcloud**
This PR moves generic operations to their respective types/ module.

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud could you move the methods to `AnyValue`, `AnyScalar` and `AnyColumn` instead? I think it would be clearer to define them there. Eventually the `ValueExpr` classes will be removed.

**Comment by kszucs** [comment]
@cpcloud needs a rebase

File: ibis/expr/types/generic.py [hunk 1]
```diff
@@ -119,7 +140,479 @@ def _repr_html_(self) -> str | None:
...
+        return ops.GroupConcat(self, sep=sep, where=where).to_expr()
+
+    def __hash__(self) -> int:
+        return super().__hash__()
+
+    def __eq__(self, other: AnyValue) -> ir.BooleanValue:
+        import ibis.expr.operations as ops
+        import ibis.expr.rules as rlz
+
+        return _binop(ops.Equals, self, rlz.any(other))
```
Comment (kszucs) [inline ibis/expr/types/generic.py:585]: Why is `rlz.any()` required here?

Comment (cpcloud) [reply ibis/expr/types/generic.py:585]: It may not be, I think was copying the existing code.

Comment (cpcloud) [reply ibis/expr/types/generic.py:585]: This is necessary for `_maybe_cast_args`. Changing that is out of scope for this PR.

**Review by kszucs — APPROVED** [review]

**Pull request merged**